python/lsst/daf/butler/_butler.py

```diff
@@ -1594,24 +1594,25 @@ def import_(self, *, directory: Optional[str] = None,
                 format: Optional[str] = None,
                 transfer: Optional[str] = None,
                 skip_dimensions: Optional[Set] = None) -> None:
-        """Import datasets exported from a different butler repository.
+        """Import datasets into this repository that were exported from a
+        different butler repository via `~lsst.daf.butler.Butler.export`.
 
         Parameters
         ----------
         directory : `str`, optional
-            Directory containing dataset files.  If `None`, all file paths
-            must be absolute.
+            Directory containing dataset files to import from. If `None`,
+            ``filename`` and all file paths specified therein must be absolute.
```

Comment:
This edit does enhance clarity for the existing behavior, thank you.

That said, do we truly need a `directory` argument? Why not require that `filename` be the full path?

I specified what turned out to be a completely superfluous `directory` along with a full path `filename`. Now that I review what the code actually does, it only used the latter. In the midst of an operation requiring two repos and a script you write to invoke `import_` and some kind of `export.yaml` file all floating around, one fewer things to specify would be welcome.

Reply:
The exported yaml isn't required to be in the same place as all the exported data files.

Reply:
Ohhhh, the exported yaml doesn't necessarily have absolute filepath information for each data file in it. Argh. So my call to `import_` *did* use the `directory` argument. I mistakenly thought the only time `directory` was used was to correctly locate the exported yaml itself. Let me think up a way to make this clearer.